src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +119,27 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  def takeRight(n: Int): Iterator[A] = {
+    val buffer = new Array[Any](n)
+    var i = 0
+    var m = 0
+    while (n > 0 && self.hasNext) {
+      buffer.update(i, self.next())
+      i = (i + 1) % n
+      if (m < n) m += 1
+    }
+    val l = m
+    new Iterator[A]() {
+      override def hasNext: Boolean = m > 0
+      override def next(): A = {
+        val value = buffer(i % l).asInstanceOf[A]
+        i = (i + 1) % l
+        m -= 1
+        value
+      }
+    }
+  }
```

Comment:
But that implementation is in `Iterable` where you can summon a new iterator for the collection in question (which means that the collection will be traversed twice). This implementation is in `Iterator` so the same double-iterator method will not work (since iterators are only traversable once). Instead a buffer of size `n` for `takeRight(n)` and `dropRight(n)` is used. 

Should I change the implementations of `takeRight`/`dropRight` in `Iterable` to use the double-iteration method from the current collections (instead of delegating to `View.TakeRight`/`View.DropRight`, which in turn delegates to this implementation in `Iterator`)?